### ISSUE TITLE
[Issue]: gemm_a8w8_blockscale (Triton) returns NaN when K/NUM_KSPLIT is not a multiple of BLOCK_SIZE_K (gfx1201 DEFAULT config, K=12800 -> vLLM garbage on R9700)

### ISSUE THREAD

**Issue by xy7ra** [ROCm/aiter#5286]
### Problem Description

`aiter.ops.triton.gemm.basic.gemm_a8w8_blockscale` returns **NaN** on gfx1201 (Radeon AI PRO R9700) whenever the selected config's split-K partition is not a whole number of `BLOCK_SIZE_K` tiles.

`configs/gfx1201/triton/gemm/gemm_a8w8_blockscale/DEFAULT.json` uses `NUM_KSPLIT: 8` for `M_LEQ_8`. The wrapper sets `SPLITK_BLOCK_SIZE = cdiv(K, NUM_KSPLIT)`. For **K = 12800** (Qwen3-32B `down_proj` at TP=2) that is **1600**, which is not a multiple of `BLOCK_SIZE_K = 128` nor of the 128-wide scale group. Inside the kernel:

- `num_k_iter = tl.cdiv(SPLITK_BLOCK_SIZE, BLOCK_SIZE_K)` = 13 → each partition over-reads 64 columns into the next one (double counting), and the last partition reads past K;
- `offs_k_scale = (pid_k * SPLITK_BLOCK_SIZE) // GROUP_K` → misaligned scale index for odd `pid_k`.

Result: NaN for M ≤ 8. K = 5120 and K = 25600 (which divide by 8·128) are correct with the same config. With `NUM_KSPLIT=1` forced, K = 12800 is correct (relerr 2e-4 vs fp32 dequant reference).

End-to-end symptom: vLLM main with `VLLM_ROCM_USE_AITER=1` on gfx1201 selecting `AiterFp8BlockScaledMMKernel` for Qwen3-32B-FP8 (block 128×128) produces `"lo lo lo lo ..."` from the first decode step. Every other linear shape in that model matched vLLM's native Triton kernel to 2e-4; only `(N=5120, K=12800)` at M ≤ 8 diverged (relerr 0.06–inf).

The bug is probably arch-independent (any arch whose config picks a ksplit that does not divide K/BLOCK_SIZE_K), gfx1201 just hits it through `DEFAULT.json`.

### Reproduction

Stack: ROCm 10.1.0a20260822 wheel nightly (HIP 7.16), torch 2.15.0a0+rocm10.1.0a20260822, triton 3.8.0+git3f6e4113, aiter main @ 456b9278, gfx1201.

```python
import torch
from aiter.ops.triton.gemm.basic.gemm_a8w8_blockscale import gemm_a8w8_blockscale as g, _get_config
fp8 = torch.float8_e4m3fn; dev = "cuda"
def inputs(M, N, K):
    x = torch.randn(M, K, device=dev, dtype=torch.float16); w = torch.randn(N, K, device=dev, dtype=torch.float16)
    xs = (x.float().view(M, K//128, 128).abs().amax(-1)/448).clamp(min=1e-6)
    xq = (x.float().view(M, K//128, 128)/xs[..., None]).view(M, K).to(fp8)
    ws = (w.float().view(N//128, 128, K//128, 128).abs().amax((1, 3))/448).clamp(min=1e-6)
    wq = (w.float().view(N//128, 128, K//128, 128)/ws[:, None, :, None]).view(N, K).to(fp8)
    ref = (xq.float().view(M, K//128, 128)*xs[..., None]).view(M, K) @ (wq.float().view(N//128, 128, K//128, 128)*ws[:, None, :, None]).view(N, K).T
    return xq, xs.contiguous(), wq, ws.contiguous(), ref
for M in (1, 2, 4, 8, 16):
    xq, xs, wq, ws, ref = inputs(M, 5120, 12800)
    cfg, _ = _get_config(M, 5120, 12800, backend="triton")
    out = g(xq, wq, xs, ws, dtype=torch.float16)
    c1 = dict(cfg); c1["NUM_KSPLIT"] = 1
    out1 = g(xq, wq, xs, ws, dtype=torch.float16, config=c1)
    err = lambda o: ((o.float()-ref).norm()/ref.norm()).item()
    print(M, cfg["NUM_KSPLIT"], err(out), err(out1))
```
Output:
```
1 8 nan 0.0002
2 8 nan 0.0002
4 8 nan 0.0002
8 8 nan 0.0002
16 1 0.0002 0.0002
```

### Proposed fix

Clamp `NUM_KSPLIT` to the largest value that divides `K / BLOCK_SIZE_K` before computing `SPLITK_BLOCK_SIZE` (or make the kernel mask the partition boundary). Minimal wrapper-side version, verified to fix the repro and the vLLM garbage (Qwen3-32B-FP8 TP=2 on 2×R9700 goes from garbage to coherent, 27.2/78.0/204.3 tok/s at 1/3/8 streams vs 25.8/71.1/182.5 with vLLM's native kernel):

```diff
@@ def gemm_a8w8_blockscale(
     if y is None and (config["NUM_KSPLIT"] == 1 or not skip_reduce):
         y = torch.empty((M, N), dtype=dtype, device=x.device)
 
+    # Split-K partitions must be a whole number of BLOCK_SIZE_K tiles (and of the
+    # 128-wide scale groups); otherwise a partition straddles a K block and the
+    # kernel produces NaN/garbage (gfx1201 DEFAULT.json: N=5120, K=12800, M<=8,
+    # ksplit=8 -> 1600 per split). Fall back to the largest ksplit that divides cleanly.
+    _bk = int(config.get("BLOCK_SIZE_K", 128))
+    _ks = int(config.get("NUM_KSPLIT", 1))
+    while _ks > 1 and (K % (_ks * _bk) != 0):
+        _ks -= 1
+    if _ks != config.get("NUM_KSPLIT", 1):
+        config = dict(config)
+        config["NUM_KSPLIT"] = _ks
     config["SPLITK_BLOCK_SIZE"] = triton.cdiv(
         K, config["NUM_KSPLIT"]
     )  # How big each split_k partition is
```

The same `SPLITK_BLOCK_SIZE = cdiv(K, NUM_KSPLIT)` pattern also exists in `gemm_a8w8_blockscale_preshuffle` in the same file.

### Operating System
Ubuntu 24.04 (container), host kernel 7.0.0-rc2

### GPU
2× AMD Radeon AI PRO R9700 (gfx1201)

### ROCm Version
10.1.0a20260822 (whl-multi-arch nightly)
